kubernetes/linux/main.sh

```diff
@@ -1,5 +1,44 @@
 #!/bin/bash
 
+# check to see if Ikey is properly formatted an instrumentation key
```

Comment:
do we need this for public cloud ? if not, where is the cloud check ?

Reply:
The cloud check is implicit. If we're in public (or another non-airgapped cloud) then the ikey will be passed in $APPLICATIONINSIGHTS_AUTH and no web requests will be made. If we're in an air-gapped cloud then $APPLICATIONINSIGHTS_AUTH will be set to a URL and the agent will try to fetch an ikey from it.

Reply:
The IKey URL is now passed in through a new environment variable - APPLICATIONINSIGHTS_AUTH_URL. It will only be set in private clouds, so checking if it is set serves as a cloud check